index.js

```diff
@@ -1,6 +1,92 @@
 let square = document.getElementById("square");
 console.log(square.style.left, square.style.top);
+const game = document.querySelector("#game")
+game.style.backgroundImage = "url(https://media.giphy.com/media/26u4o0RRnUxq3vuKI/giphy.gif)"
 
 document.addEventListener("keydown", function(e) {
-  console.log(e.key)
+  // left movement
+  const moveLeft = square.offsetLeft
+  const moveTop = square.offsetTop
+  if(e.key === 'ArrowLeft' && square.style.left !== `0px`) {
+    let move = (moveLeft - 10) + "px"
+    square.style.left = move
+  } else if (e.key === 'ArrowRight' && square.style.left !== `300px`) { //right movement
+    let move = (moveLeft + 10) + "px"
+    square.style.left = move
+  } else if (e.key === 'ArrowUp' && square.style.top !== `0px`) { //upward movement
+    let move = (moveTop - 10) + "px"
+    square.style.top = move
+  } else if (e.key === 'ArrowDown' && square.style.top !== `300px`) {  //downward movement
+    let move = (moveTop + 10) + "px"
+    square.style.top = move
+  };
+
+
 });
+
+document.addEventListener("click", function(a) {
+  if(a.target === square) {
+    let randomNum = Math.floor(Math.random() * 255)
+    let randomNum1 = Math.floor(Math.random() * 255)
+    let randomNum2 = Math.floor(Math.random() * 255)
+    square.style.backgroundColor = `rgb(${randomNum1}, ${randomNum2}, ${randomNum})`
+  }
+})
```

Comment:
It would be a better practice to target the `square` directly instead of putting the listener on the document:
```js
square.addEventListener("click", function(a) {
    let randomNum = Math.floor(Math.random() * 255)
    let randomNum1 = Math.floor(Math.random() * 255)
    let randomNum2 = Math.floor(Math.random() * 255)
    square.style.backgroundColor = `rgb(${randomNum1}, ${randomNum2}, ${randomNum})`
})
```